internal/runbits/buildscript/buildscript.go

```diff
@@ -0,0 +1,92 @@
+package buildscript
+
+import (
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/internal/output"
+	bpModel "github.com/ActiveState/cli/pkg/platform/api/graphql/model/buildplanner"
+	"github.com/ActiveState/cli/pkg/platform/authentication"
+	"github.com/ActiveState/cli/pkg/platform/runtime/buildscript"
+	"github.com/ActiveState/cli/pkg/platform/runtime/model"
+	"github.com/ActiveState/cli/pkg/project"
+	"github.com/go-openapi/strfmt"
+)
+
+func getRemoteBuildScript(proj *project.Project, customCommit *strfmt.UUID, auth *authentication.Auth) (*bpModel.BuildScript, error) {
+	bp := model.NewBuildPlanner(auth)
+	commitID := proj.CommitUUID()
+	if customCommit != nil {
+		commitID = *customCommit
+	}
+	return bp.GetBuildScript(proj.Owner(), proj.Name(), commitID.String())
+}
+
+// Sync synchronizes the local build script with the remote one.
+// If a commit ID is given, a local mutation has occurred (e.g. added a package, pulled, etc.), so
+// pull in the new build script. Otherwise, if there are local build script changes, create a new
+// commit with them in order to update the remote one.
+func Sync(proj *project.Project, commitID *strfmt.UUID, out output.Outputer, auth *authentication.Auth) error {
+	logging.Debug("Synchronizing local build script")
+	file, err := buildscript.Get(proj.Dir())
+	if err != nil && !buildscript.IsDoesNotExistError(err) {
+		return errs.Wrap(err, "Could not get local build script")
+	}
+
+	script, err := getRemoteBuildScript(proj, commitID, auth)
```

Comment:
We should update the naming of things in this PR. The BuildScript refers to the local order file whereas the BuildExpression is what the buildplanner sends and receives. So this function should be `getBuildExpression`

Reply:
I've done some more thorough renaming here and elsewhere. Thanks for the feedback.